lib/monad.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+// id :: a -> a
+function id(x) {
+  return x;
+}
+
+// curry :: ((a, b) -> c) -> a -> b -> c
+function curry(f, ...args1) {
```

Comment:
We have 3 implementation:
- `api.common.curry`
- `lib/monad.js/curry`
- `(f, x) => (...args) => f(x, ...args)`

Please make research which will be classical and which more usable